### PR TITLE
Part 1 & 2 Submission

### PR DESCRIPTION
**Pull request by kangraye**
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment? 2 hours

Please fill in one error and/or error message you received while working on this assignment. Nil.

What part of the assignment did you spend the most time on? Figuring out how to upload to GitHub. (Not sure if this is the right way to submit the part 2 submission. Do I just override part 1?)

Comfort Level (1-5): 5

Completeness Level (1-5): 4 (Did not add in the emojis as part of the string output)

What did you think of this deliverable? Quite a lot of lines of codes. Did not think of re-factorizing earlier and kind of lazy to do it now that there are so many lines.

Is there anything in this code that you feel pleased about? I was able to combine all modes of play and (I think) all modes are working.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -1,4 +1,194 @@
+var generateOutcome = function () {
+  var randomInt = Math.floor(Math.random() * 3);
+  var map = { 0: "scissors", 1: "paper", 2: "stone" };
```
Comment (emily-zhangg) [inline script.js:3]: nice! really like the use of objects instead of if statements to get the value associated with the number generated

File: script.js [hunk 2]
```diff
@@ -1,4 +1,194 @@
...
+    } else if (
+      (userChoice == "scissors" && computerOutcome == "paper") ||
+      (userChoice == "paper" && computerOutcome == "stone") ||
+      (userChoice == "stone" && computerOutcome == "scissors")
+    ) {
+      outcome = win;
+      wins = wins + 1;
+      games = games + 1;
+    } else {
+      outcome = invalid;
```
Comment (emily-zhangg) [inline script.js:82]: perhaps you could make a helper function that validates the user input from the start so that over here in your if statements, you could let anything that does not fulfil the win and draw condition be set to lose. This will remove the need for you to list out all the losing conditions too.

File: script.js [hunk 3]
```diff
@@ -1,4 +1,194 @@
...
+    var outcome = invalid;
+
+    // If else logic for normal mode
+    if (
+      (userChoice == "scissors" && computerOutcome == "stone") ||
+      (userChoice == "paper" && computerOutcome == "scissors") ||
+      (userChoice == "stone" && computerOutcome == "paper")
+    ) {
+      outcome = lose;
+      loss = loss + 1;
```
Comment (emily-zhangg) [inline script.js:68]: a convenient syntax for this would be loss +=1 which would be the same as loss = loss +1

File: script.js [hunk 4]
```diff
@@ -1,4 +1,194 @@
...
+  myOutputValue =
+    `You chose ${userChoice}. <br>The computer generated ${computerOutcome}<br><br>` +
+    outcome +
+    `<br><br>So far ${playerName}, your record is as follows. ${evaluation}<br>
+    Games: ${games}<br>
+    Wins: ${wins}<br>
+    Loss: ${loss}<br>
+    Draw: ${games - wins - loss}<br><br>
+    Now you can type "scissors", "paper", or "stone" to play another round!`;
   return myOutputValue;
```
Comment (emily-zhangg) [inline script.js:193]: I liked that the code was readable due to proper variable namings as well as adequate comments. However, I think that it could be easier to read if you pull out your logic intensive game modes as separate helper functions that you call within the main functions. This is because when we write functions, a general rule of thumb will be to not write anything that spans more than 3/4 a screen. This is so that it is easier to understand for others and for yourself too when you go back to read  your code. By abstracting the logic into helper functions, you would no longer have to read through each line of specific logic to understand what the function does.

**Review by emily-zhangg — COMMENTED** [review]
great job! I really like the use of objects in your generateOutcome function. Generally, I think you have a good grasp of the concepts used here as well as the logic required.
